rfcs/20200618-headers/README.md

```diff
@@ -0,0 +1,123 @@
+# Changes in the oneDNN header install locations and namespace
+
+## Introduction
+
+New headers structure and namespaces are needed for cross-component and oneAPI
+branding alignment. The request from the oneAPI program is to put oneDNN
+headers under the `oneapi` and also to provide a corresponding namespace. The
+changes will have to be reflected in the
+[spec](https://github.com/oneapi-src/oneAPI-spec/tree/master/source/elements/oneDNN).
+
+The main concern is how to keep backward compatibility with the existing code.
+The secondary concern is maintenance burden.
+
+## Proposal
+
+The following changes are to be implemented in the
+[`dev-v2`](https://github.com/oneapi-src/onednn/tree/dev-v2) branch but will
+eventually be part of master.
+
+### Namespace changes
+
+oneDNN should start providing two namespaces (C++ only): `dnnl::` and
+`oneapi::dnnl::`. Either can be an alias. To minimize the changes, the
+proposal is to alias the long one:
+
+```c++
+namespace dnnl { /* the usual stuff */ }
+
+namespace oneapi { namespace dnnl = ::dnnl; }
+```
+
+The new namespace must be reflected in the
+[spec](https://github.com/oneapi-src/oneAPI-spec/tree/master/source/elements/oneDNN).
+There is no recommendation on whether longer or shorted namespace should be
+used in the spec yet; this is outside of the scope of this RFC.
+
+### Installation paths changes
+
+oneDNN is free to decide on the header structure under `include/oneapi`.
```

Comment:
Will we have similar folder structure for installed libs?

Reply:
Not sure I got the question completely, but oneDNN will have the same structure for installed library and in the repo.

Reply:
I meant a path like `lib64/oneapi/dnnl/libdnnl.so`?

Reply:
No, this is only about header files, not the library files.